src/main/scala/com/fasterxml/jackson/module/scala/modifiers/ScalaTypeModifier.scala

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.module.scala.modifiers
+
+import java.lang.reflect.{ParameterizedType, Type}
+
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.`type`._
+import com.fasterxml.jackson.module.scala.JacksonModule
+
+class ScalaTypeModifier extends TypeModifier {
+  override def modifyType(javaType: JavaType,
+                          jdkType: Type,
+                          context: TypeBindings,
+                          typeFactory: TypeFactory): JavaType = {
+
+
+    if (javaType.isTypeOrSubTypeOf(classOf[collection.Map[_, _]])) {
+      MapLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0), javaType.containedTypeOrUnknown(1))
```

Comment:
@nbauernfeind Thanks; I would be interested in such behavior if it occurs (wouldn't be altogether surprised), and perhaps how to work around that. But good to have a warning that there be possibly dragons.